spec/rubocop/cop/style/case_condition_repeat_spec.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Style::CaseConditionRepeat do
+  subject(:cop) { described_class.new }
+
+  let(:offense_source) do
+    ['case x',
+     'when false',
+     '  first_method',
+     'when true',
+     '  second_method',
+     'when false',
+     '  third_method',
+     'end']
+  end
+
+  let(:offense_source2) do
+    ['case x',
+     'when false',
+     '  first_method',
+     'when false',
+     '  second_method',
+     'end']
+  end
+
+  let(:offense_source3) do
+    ['case x',
+     'when false',
+     '  first_method',
+     'when true',
+     '  second_method',
+     'when false',
+     '  third_method',
+     'when true',
+     '  fourth_method',
+     'end']
+  end
+
+  let(:offense_source4) do
+    ['case x',
+     'when a, b',
+     '  first_method',
+     'when b, a',
+     '  second_method',
+     'end']
+  end
+
+  let(:offense_source5) do
+    ['case x',
+     'when a && b',
+     '  first_method',
+     'when a && b',
+     '  second_method',
+     'end']
+  end
+
+  let(:accept_source) do
+    ['case x',
+     'when false',
+     '  first_method',
+     'end']
+  end
+
+  let(:accept_source2) do
+    ['case x',
+     'when false',
+     '  first_method',
+     'when true',
+     '  second_method',
+     'end']
+  end
+
+  let(:accept_source3) do
+    ['case x',
+     'when false',
+     '  method_name',
+     'when true',
+     '  second_method',
+     'else',
+     '  third_method',
+     'end']
+  end
+
+  let(:accept_source4) do
+    ['case x',
+     'when something && another && other',
+     '  first_method',
+     'when something && another',
+     '  second_method',
+     'end']
+  end
+
+  it 'registers an offense for repeated case conditionals' do
+    inspect_source(cop, offense_source)
```

Comment:
I would much prefer to have the source juxtaposed with the assertion, as now I have to scroll up and down, searching for very similarly names `let` assignments. 😞